bodo/pandas/frame.py

```diff
@@ -831,16 +819,12 @@ def merge(
         empty_join_out.columns = [
             c + str(i) for i, c in enumerate(empty_join_out.columns)
         ]
-        # We want to avoid having self appear on the rhs since the unique_ptr
-        # to the duckdb plan will delete it on the lhs first before it processes the rhs
-        # TODO [BSE-4865]: check right._plan for self recursively.
-        right_plan = deepcopy(right._plan) if self is right else right._plan
 
         planComparisonJoin = LazyPlan(
             "LogicalComparisonJoin",
             empty_join_out,
             self._plan,
-            right_plan,
+            right._plan,
```

Comment:
How was this tested?  The join tests we have are not sufficient.  In one of my tpch branches I changed this to unilateral deepcopy.  Before this goes in, could you create a branch of this and merge with my tpch2 branch (the one from the current PR) and run the tpch queries that are supposed to pass in the spreadsheet.

Reply:
We have a bunch of join tests that failed without this change so it's exercised.